### PR TITLE
HAWQ-1297. Make PXF install ready from source code

### PR DESCRIPTION
**Pull request by shivzone**

### CODE REVIEW

**Comment by shivzone** [comment]
https://issues.apache.org/jira/browse/HAWQ-1297

File: pxf/pxf-service/src/scripts/pxf-service [hunk 1]
```diff
@@ -193,39 +210,83 @@ function doInit()
...
+    pushd $instance_root/$instance_name/webapps || return 1
+    rm -rf pxf
+    mkdir pxf
+    cd pxf
+    unzip -q ../pxf.war
+    popd
+
+    context_file=$instance_root/$instance_name/webapps/pxf/META-INF/context.xml
+    cat $context_file | \
+    sed  -e "s:classpathFiles=\"[a-zA-Z0-9\/\;.-]*\":classpathFiles=\"$PXF_HOME\/conf\/pxf-private.classpath\":" \
```
Comment (sansanichfb) [inline pxf/pxf-service/src/scripts/pxf-service]: Why not combine two steps into one?

File: pxf/pxf-service/src/main/resources/pxf-private.classpath [hunk 2]
```diff
@@ -0,0 +1,67 @@
+##################################################################
```
Comment (sansanichfb) [inline pxf/pxf-service/src/main/resources/pxf-private.classpath]: Would RAT complain about absence of Apache header?

File: pxf/pxf-service/src/scripts/pxf-service [hunk 3]
```diff
@@ -40,6 +34,29 @@ else
...
+else
+    # OSS/Source code based setup
+    pxf_root=$PXF_HOME/lib
+    tomcat_root=$PXF_HOME/apache-tomcat
+    tomcat_templates=$PXF_HOME/tomcat-templates
+    instance_root=$PXF_HOME
+    instance_owner=$pxf_user
+fi
+
+curl=`which curl`
```
Comment (kavinderd) [inline pxf/pxf-service/src/scripts/pxf-service]: Maybe check that this command returns a non-null value

File: pxf/pxf-service/src/scripts/pxf-service [hunk 4]
```diff
@@ -193,39 +210,83 @@ function doInit()
 		return 0
 	fi
 
-	createInstance || return 1
+    createInstance || return 1
```
Comment (kavinderd) [inline pxf/pxf-service/src/scripts/pxf-service]: indentation

File: pxf/pxf-service/src/scripts/pxf-service [hunk 5]
```diff
@@ -193,39 +210,83 @@ function doInit()
...
-	command=$1
-
 	instanceExists
 	if [ $? -ne 0 ]; then
 		echo ERROR: cant find PXF instance, maybe call init?
 		return 1
 	fi
+	patchWebapp || return 1
+    commandWebapp start || return 1
+    checkWebapp 300 || return 1
```
Comment (kavinderd) [inline pxf/pxf-service/src/scripts/pxf-service]: Indentation of line 273, 274 doesn't match rest of the function

File: pxf/pxf-service/src/scripts/pxf-service [hunk 6]
```diff
@@ -193,39 +210,83 @@ function doInit()
...
+		echo ERROR: cant find PXF instance, maybe call init?
 		return 1
-	fi 
-	popd
-	
-	if [ "$command" = "start" ]; then
-		# try to connect for 5 minutes
-		checkWebapp 300 || return 1
 	fi
+    commandWebapp stop || return 1
```
Comment (kavinderd) [inline pxf/pxf-service/src/scripts/pxf-service]: Indentation off compared to the rest of the function

File: pxf/pxf-service/src/scripts/pxf-service [hunk 7]
```diff
@@ -193,39 +210,83 @@ function doInit()
...
+#
+# doStart handles stop command
+# command is executed as the user $pxf_user
+#
+#
+function doStop()
+{
+	instanceExists
 	if [ $? -ne 0 ]; then
+		echo ERROR: cant find PXF instance, maybe call init?
```
Comment (kavinderd) [inline pxf/pxf-service/src/scripts/pxf-service]: quote the string just to be safe

**Pull request closed without merging**